src/lib/swapper/swappers/LifiSwapper/utils/getNetworkFeeCryptoBaseUnit/getNetworkFeeCryptoBaseUnit.ts

```diff
@@ -0,0 +1,65 @@
+import type { Step } from '@lifi/sdk'
+import type { ChainId } from '@shapeshiftoss/caip'
+import { KnownChainIds } from '@shapeshiftoss/types'
+import type { AbiItem } from 'web3-utils'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import { bn, bnOrZero } from 'lib/bignumber/bignumber'
+import { SwapError, SwapErrorType } from 'lib/swapper/api'
+import { isEvmChainAdapter } from 'lib/utils'
+import { getWeb3InstanceByChainId } from 'lib/web3-instance'
+
+import { getLifi } from '../getLifi'
+
+export const getNetworkFeeCryptoBaseUnit = async ({
+  chainId,
+  lifiStep,
+}: {
+  chainId: ChainId
+  lifiStep: Step
+}) => {
+  const lifi = getLifi()
+  const adapter = getChainAdapterManager().get(chainId)
+
+  if (adapter === undefined || !isEvmChainAdapter(adapter)) {
+    throw new SwapError('[getNetworkFeeCryptoBaseUnit] invalid chain adapter', {
+      code: SwapErrorType.TRADE_QUOTE_FAILED,
+      details: { chainId },
+    })
+  }
+
+  const { transactionRequest } = await lifi.getStepTransaction(lifiStep)
+  const { value, from, to, data, gasLimit, gasPrice, maxFeePerGas } = transactionRequest ?? {}
+
+  if (value === undefined || from === undefined || to === undefined || data === undefined) {
```

Comment:
We're dealing with `string | number | bigint | BigNumber | Bytes | undefined` type here, I believe we may be able to use the `!` operator on most of these - may have to double check whether 0-value is returned as `0`, `'0'`, a 0 BN instance though